lib/puppet/application/device.rb

```diff
@@ -319,15 +327,12 @@ def main
             end
           else
```

Comment:
if `options[:libdir]`, we'll enter this block, but we won't have gone through the above block first. Will we still have `ssl_context` defined in that case? My gut says no. Also, will `ssl_context` be available outside of the block that starts on line 272?

Reply:
Good point. @DavidS if `--libdir` is specified, then I assume we want to skip pluginsync, but we do want to apply the configuration to the target devices (in which case we'll need to perform ssl bootstrap earlier)?

Reply:
I added another commit that matches my comment above.